### PR TITLE
Update plugins

### LINKED ISSUE #1340: Unexpected Fetch latest versions failure
The automatic Fetch latest versions workflow failed, please investigate:
https://github.com/bufbuild/plugins/actions/runs/9875732168/attempts/1

### PR DESCRIPTION
**Pull request by mfridman**
Includes

- connectrpc/kotlin:v0.7.0
- community/mercari-grpc-federation:v1.0.3
- apple/swift:v1.27.0 (with new patch), related:
  - https://github.com/apple/swift-protobuf/pull/1539
  - https://github.com/apple/swift-protobuf/pull/1221

Fix #1340

### CODE REVIEW

File: plugins/apple/swift/v1.27.0/extramoduleimports.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ 
+diff --git a/Sources/protoc-gen-swift/GeneratorOptions.swift b/Sources/protoc-gen-swift/GeneratorOptions.swift
+index 2635cb8d..58a75681 100644
+--- a/Sources/protoc-gen-swift/GeneratorOptions.swift
++++ b/Sources/protoc-gen-swift/GeneratorOptions.swift
+@@ -49,6 +49,7 @@ class GeneratorOptions {
+     }
+   }
+ 
++  let extraModuleImports: [String]
```
Comment (mfridman) [inline plugins/apple/swift/v1.27.0/extramoduleimports.patch:28]: Moved some of the constants and variables towards the "top" to minimize conflicts.

**Comment by mfridman** [comment]
apple/swift generated code diff between v1.26.0 and v1.27.0

buf_connect_demo_eliza_v1_eliza.pb.swift

```diff
38c38
< public struct Buf_Connect_Demo_Eliza_V1_SayRequest {
---
> public struct Buf_Connect_Demo_Eliza_V1_SayRequest: Sendable {
51c51
< public struct Buf_Connect_Demo_Eliza_V1_SayResponse {
---
> public struct Buf_Connect_Demo_Eliza_V1_SayResponse: Sendable {
64c64
< public struct Buf_Connect_Demo_Eliza_V1_ConverseRequest {
---
> public struct Buf_Connect_Demo_Eliza_V1_ConverseRequest: Sendable {
77c77
< public struct Buf_Connect_Demo_Eliza_V1_ConverseResponse {
---
> public struct Buf_Connect_Demo_Eliza_V1_ConverseResponse: Sendable {
90c90
< public struct Buf_Connect_Demo_Eliza_V1_IntroduceRequest {
---
> public struct Buf_Connect_Demo_Eliza_V1_IntroduceRequest: Sendable {
103c103
< public struct Buf_Connect_Demo_Eliza_V1_IntroduceResponse {
---
> public struct Buf_Connect_Demo_Eliza_V1_IntroduceResponse: Sendable {
115,123d114
< #if swift(>=5.5) && canImport(_Concurrency)
< extension Buf_Connect_Demo_Eliza_V1_SayRequest: @unchecked Sendable {}
< extension Buf_Connect_Demo_Eliza_V1_SayResponse: @unchecked Sendable {}
< extension Buf_Connect_Demo_Eliza_V1_ConverseRequest: @unchecked Sendable {}
< extension Buf_Connect_Demo_Eliza_V1_ConverseResponse: @unchecked Sendable {}
< extension Buf_Connect_Demo_Eliza_V1_IntroduceRequest: @unchecked Sendable {}
< extension Buf_Connect_Demo_Eliza_V1_IntroduceResponse: @unchecked Sendable {}
< #endif  // swift(>=5.5) && canImport(_Concurrency)
< 
```

File: plugins/apple/swift/v1.27.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# syntax=docker/dockerfile:1.7
```
Comment (pkwarren) [inline plugins/apple/swift/v1.27.0/Dockerfile:1]:
```suggestion
# syntax=docker/dockerfile:1.8
```

File: plugins/connectrpc/kotlin/v0.7.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# syntax=docker/dockerfile:1.7
```
Comment (pkwarren) [inline plugins/connectrpc/kotlin/v0.7.0/Dockerfile:1]:
```suggestion
# syntax=docker/dockerfile:1.8
```

File: plugins/connectrpc/kotlin/v0.7.0/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+FROM debian:bookworm-20240701 AS build
+RUN apt-get update \
+ && apt-get install -y curl
+WORKDIR /app
+RUN curl -fsSL -o /app/protoc-gen-connect-kotlin.jar https://repo1.maven.org/maven2/com/connectrpc/protoc-gen-connect-kotlin/0.7.0/protoc-gen-connect-kotlin-0.7.0.jar
+
+FROM gcr.io/distroless/java17-debian12:latest@sha256:c8c2c236c63d7652b707a05b02d0259db91d5feb357e9dc53942f73c2e001088
```
Comment (pkwarren) [inline plugins/connectrpc/kotlin/v0.7.0/Dockerfile:8]: We should squash layers here similar to other builds.

File: plugins/connectrpc/kotlin/v0.7.0/buf.plugin.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+version: v1
+name: buf.build/connectrpc/kotlin
+plugin_version: v0.7.0
+source_url: https://github.com/connectrpc/connect-kotlin
+integration_guide_url: https://connectrpc.com/docs/kotlin/getting-started
+description: Idiomatic gRPC & Connect RPCs for Kotlin.
+deps:
+  - plugin: buf.build/protocolbuffers/kotlin:v26.1
```
Comment (pkwarren) [inline plugins/connectrpc/kotlin/v0.7.0/buf.plugin.yaml:8]:
```suggestion
  - plugin: buf.build/protocolbuffers/kotlin:v27.2
```

We've updated to 27.2 in connect-kotlin.

Comment (mfridman) [reply plugins/connectrpc/kotlin/v0.7.0/buf.plugin.yaml:8]: Phew, I was wondering when we'd bump this.

File: plugins/apple/swift/v1.27.0/Dockerfile (hunk 2 above)
Comment (mfridman) [reply plugins/apple/swift/v1.27.0/Dockerfile:1]: I'm going to automate this once and for all 😛

File: plugins/connectrpc/kotlin/v0.7.0/Dockerfile (hunk 4 above)
Comment (mfridman) [reply plugins/connectrpc/kotlin/v0.7.0/Dockerfile]: Updated in 1ea0fb87, in case you want to review that separately

**Review by pkwarren — APPROVED** [review]

**Pull request merged**